openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -362,6 +421,61 @@ def pre_requisite_courses(self):
         return json.loads(self._pre_requisite_courses_json)
 
     @classmethod
+    def get_select_courses(cls, course_keys):
+        """
+        Returns CourseOverview objects for the given course_keys.
+        """
+        course_overviews = []
+
+        log.info('Generating course overview for %d courses.', len(course_keys))
+        log.debug('Generating course overview(s) for the following courses: %s', course_keys)
+
+        for course_key in course_keys:
+            try:
+                course_overviews.append(CourseOverview.get_from_id(course_key))
+            except Exception as ex:  # pylint: disable=broad-except
+                log.exception(
+                    'An error occurred while generating course overview for %s: %s',
+                    unicode(course_key),
+                    ex.message,
+                )
+
+        log.info('Finished generating course overviews.')
+
+        return course_overviews
+
+    @classmethod
+    def get_all_courses(cls, force_reseeding=False):
+        """
+        Returns all CourseOverview objects in the database.
+
+        Arguments:
+            force_reseeding (bool): Optional parameter.
+                If True, the modulestore is used as the source of truth for
+                the list of courses, even if the CourseOverview table was
+                previously seeded.
+
+                If False, the list of courses is retrieved from the
+                CourseOverview table if it was previously seeded, falling
+                back to the modulestore if it wasn't seeded.
+        """
+        if force_reseeding or not CourseOverviewGeneratedHistory.objects.first():
+            # Seed the CourseOverview table with data for all
+            # courses in the system.
+            course_keys = [course.id for course in modulestore().get_courses()]
+            course_overviews = cls.get_select_courses(course_keys)
+            CourseOverviewGeneratedHistory.objects.create(num_courses=len(course_overviews))
```

Comment:
The scenarios here are:

1. edx.org, which doesn't call this method for its LMS course catalog page since it uses Drupal for the catalog.  However, if someone accesses the Course Catalog API on edx.org, then this method would be exercised.  In that case, we should pre-empt any issues by having devOps call the generate_course_overview management command when this is released.

2. open edX instance that calls this method to display its Course Catalog on the LMS index page.  In that case, they were already calling `modulestore().get_courses()` and already instantiating all the course modules in the system to render that page.  With this code, they can either preemptively run the management command or lazily populate the cache the first time the Course Catalog view is rendered.

3. sandboxes connected to prod-clone would be in a similar boat to #2, except since they have a lot of courses, it's better if the management command is preemptively run.

Am I missing something?